library/src/main/java/com/novoda/downloadmanager/lib/DownloadTask.java

```diff
@@ -587,11 +544,17 @@ private void transferData(State state, InputStream in, OutputStream out) throws
 
         State newState = dataTransferer.transferData(state, in);
         handleEndOfStream(newState);
+        LLog.v("end transfer data");
     }
 
     private final NotifierWriter.WriteChunkListener checkOnWrite = new NotifierWriter.WriteChunkListener() {
         @Override
         public void chunkWritten(FileDownloadInfo downloadInfo) throws StopRequestException {
+            if (clock.intervalLessThan(Clock.Interval.ONE_SECOND)) {
```

Comment:
Write to disc in chunks can happen really fast, in order to do not hammer the client we only check once every second.